src/lib.rs

```diff
@@ -0,0 +1,320 @@
+#![crate_type = "staticlib"]
+
+use libc::{c_char, c_int, c_void, size_t, ssize_t};
+use std::io::{Cursor, Read, Write};
+use std::slice;
+use std::{ffi::CStr, sync::Arc};
+use std::{io::ErrorKind::ConnectionAborted, mem};
+
+use rustls::{ClientConfig, ClientSession, Session};
+
+type CrustlsResult = c_int;
+
+pub const CRUSTLS_OK: c_int = 0;
+pub const CRUSTLS_ERROR: c_int = 1;
```

Comment:
Could this be a Rust enum that has an appropriate `Into` implementation to convert into `c_int`?

Reply:
I started on this and could not get cbindgen to generate the definitions right. I'm sure it's possible, but I'd like to punt to another PR.